exporter/elasticsearchexporter/internal/metricgroup/hasher.go

```diff
@@ -0,0 +1,133 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metricgroup // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/metricgroup"
+
+import (
+	"encoding/binary"
+	"hash/fnv"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/datapoints"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/elasticsearch"
+)
+
+// HashKey is a struct for comparing data point identity.
+type HashKey struct {
+	resourceHash uint32
+	scopeHash    uint32
+	dpHash       uint32
+}
```

Comment:
My initial idea was to not have `{ecs, otel}DataPointHasher` at all and build the hash update logic in the `HashKey` but I think we can discuss that in the follow-up PR when the pdatautil has the method for hashing map with exclude keys.

Reply:
See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39309#discussion_r2041853809

Reply:
Hmm, this is a bit different than that comment. I was thinking that once we have the hasher methods in the pdatautil, we could remove the `DataPointHasher` completely and build an API around HashKey (assuming pdata is hierarchical) to calculate the hashes. We can discuss this at a later point.